### PR TITLE
Optimizing HROM Setup Performance

### PR DESCRIPTION
**Pull request by SADPR**
## Summary

This PR introduces substantial improvements to the setup process for HROMs in Kratos. It replaces the previous dictionary-based storage for ROM elements and weights with a more efficient list-based approach. This change addresses performance bottlenecks identified during the handling of large datasets, where the former method was projected to take impractically long execution times (up to weeks for extensive cases like the motor setup; now, it takes a few seconds).

### Replaced Data Structures
- **Old Implementation**: Used a dictionary to store weights and indices for ROM elements and conditions. This method was not only memory-intensive but also significantly slower for large data sets due to the overhead associated with dictionary operations in Python and C++.

- **New Implementation**: Shifts to using lists and numpy arrays that streamline the data handling and improve the execution speed. The new method directly loads and processes `.npy` files for element and condition IDs, and weights into numpy arrays, facilitating faster access and operations.

### Additional Enhancements
- **Minor Bug Fixes**: Addressed a few default minor bugs that were affecting the efficiency and correctness of the HROM setups.

- **SVD Type Setting**: Introduced a new setting (`svd_type`) that allows users to choose between using the randomized SVD (rSVD) and numpy's SVD. 

- **HROM Creation Test**: Added a test for the HROM creation process, an essential part of our testing suite that was previously missing.

### CODE REVIEW

File: applications/RomApplication/python_scripts/rom_analysis.py [hunk 1]
```diff
@@ -167,7 +175,7 @@ def ModifyAfterSolverInitialize(self):
                                 aux[j,i] = petrov_galerkin_nodal_modes[node_id][j][i].GetDouble()
                         node.SetValue(KratosROM.ROM_LEFT_BASIS, aux)
 
-            elif self.rom_parameters["rom_format"].GetString() == "numpy":
```
Comment (Rbravo555) [inline applications/RomApplication/python_scripts/rom_analysis.py:172]: let us name the options:
"numpy_svd", "numpy_rsvd", later on we could add: "dislib_tsqr_svd", "dask_rsvd", and so on

File: applications/RomApplication/tests/hrom_training_utility_test_files/couette_flow_test.mdpa [hunk 2]
```diff
@@ -0,0 +1,1463 @@
+Begin ModelPartData
```
Comment (Rbravo555) [inline applications/RomApplication/tests/hrom_training_utility_test_files/couette_flow_test.mdpa]: Why arent we refering to the original file, therefore avoiding copying it in the RomApp folder? I mean:
"input_filename" : "../../../../FluidDynamicsApplication/tests/CouetteFlowTest/couette_flow_test"

**Review by Rbravo555 — COMMENTED** [review]
minor comments to be addressed

File: applications/RomApplication/tests/hrom_training_utility_test_files/couette_flow_test.mdpa (hunk 2 above)
Comment (SADPR) [reply applications/RomApplication/tests/hrom_training_utility_test_files/couette_flow_test.mdpa]: Change done.

**Comment by Rbravo555** [comment]
Not passing the tests. Is this related to the use of the model part from the CFDApp?

**Comment by Rbravo555** [comment]
as a TODO for future PRs, we should create an SVD class, so that we can move there the svds implemented in this PR in the hrom_training_untility.py

**Review by Rbravo555 — APPROVED** [review]

**Pull request merged**